internal/controller/dnsrecord_controller.go

```diff
@@ -358,14 +368,34 @@ func (r *DNSRecordReconciler) SetupWithManager(mgr ctrl.Manager, maxRequeue, val
 			}
 			return toReconcile
 		})).
+		Watches(&v1alpha1.DNSHealthCheckProbe{}, handler.EnqueueRequestsFromMapFunc(func(ctx context.Context, o client.Object) []reconcile.Request {
+			logger := log.FromContext(ctx)
+			probe, ok := o.(*v1alpha1.DNSHealthCheckProbe)
+			if !ok {
+				logger.V(1).Info("unexpected object type", "error", fmt.Sprintf("%T is not a *v1alpha1.DNSHealthCheckProbe", o))
+			}
+
+			records := &v1alpha1.DNSRecordList{}
```

Comment:
I think here we could do a get based on the owner ref on the probe? We only have one probe here and that probe can only belong to one record right?